src/averageAges.js

```diff
@@ -55,7 +94,33 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const peopleNew = [];
+  let bornChildren = 0;
+  let peopleNew2 = {};
+
+  for (const i of people) {
+    mother = i.mother;
+    bornChildren = i.born;
+
+    if (people.find(callback) && (i.sex === 'm') && (onlyWithSon === true)) {
+      peopleNew2 = people.find(callback);
+      peopleNew2.bornChildren = bornChildren;
+      peopleNew.push({ ...peopleNew2 });
+    } else {
+      if (people.find(callback) && (onlyWithSon !== true)) {
+        peopleNew2 = people.find(callback);
+        peopleNew2.bornChildren = bornChildren;
+        peopleNew.push({ ...peopleNew2 });
+      }
+    }
+  }
+
+  function callbackSum2(summa, n) {
```

Comment:
Consider renaming the function 'callbackSum2' to a more descriptive name like 'sumAgeDifferenceWithChildren'.